modules/core/src/main/scala/org/scalasteward/core/gitlab/http4s/Http4sGitlabApiAlg.scala

```diff
@@ -116,15 +141,49 @@ class Http4sGitLabApiAlg[F[_]: MonadThrowable](
 
   def createPullRequest(repo: Repo, data: NewPullRequestData): F[PullRequestOut] = {
     val targetRepo = if (doNotFork) repo else repo.copy(owner = user.login)
-    for {
+    val mergeRequest = for {
       projectId <- client.get[ProjectId](url.repos(repo), modify(repo))
       payload = MergeRequestPayload(url.encodedProjectId(targetRepo), projectId.id, data)
-      res <- client.postWithBody[PullRequestOut, MergeRequestPayload](
+      res <- client.postWithBody[MergeRequestOut, MergeRequestPayload](
         url.mergeRequest(targetRepo),
         payload,
         modify(repo)
       )
     } yield res
+
+    def waitForMergeRequestStatus(internalId: Int, retries: Int = 10): F[MergeRequestOut] =
+      client
+        .get[MergeRequestOut](url.existingMergeRequest(repo, internalId), modify(repo))
+        .flatMap {
+          case mr if (mr.mergeStatus =!= GitlabMergeStatus.Checking) => monad.pure(mr)
+          case _ if (retries > 0)                                    => waitForMergeRequestStatus(internalId, retries - 1)
+          case other                                                 => monad.pure(other)
+        }
+
+    val updatedMergeRequest =
+      if (!mergeWhenPipelineSucceeds)
+        mergeRequest
+      else
+        mergeRequest
+          .flatMap(mr => waitForMergeRequestStatus(mr.iid))
+          .flatMap {
+            case mr if (mr.mergeStatus === GitlabMergeStatus.CanBeMerged) =>
+              for {
+                _ <- logger.info(s"Setting ${mr.webUrl} to merge when pipeline succeeds")
+                res <- client.put[MergeRequestOut](
+                  url.mergeWhenPiplineSucceeds(repo, mr.iid),
+                  modify(repo)
+                )
+              } yield res
+            case mr =>
+              logger.info(s"Unable to automatically merge ${mr.webUrl}").map(_ => mr)
+          }
+          .recoverWith { case UnexpectedResponse(_, _, _, status, _) =>
+            logger
+              .warn(s"Unexpected gitlab response setting auto merge: $status")
+              .flatMap(_ => mergeRequest)
```

Comment:
I'm not sure I understand this. `mergeRequest` has already been used on line 167 and if that fails, I don't think it will succeed on a second try here. If the first try succeeded and the merge request has already been created but `waitForMergeRequestStatus` or `client.put[MergeRequestOut](...)` throws an `UnexpectedResponse` using `mergeRequest` here again seems to be unnecessary - and I hope it will not create a second merge request. It makes more sense to me to only recover from `UnexpectedResponse` in the combination of `waitForMergeRequestStatus` and `client.put[MergeRequestOut](...)` (i.e. lines 168 to 180).

Reply:
Thanks for the review @fthomas! All of this happens after the merge request was created in attempt to just update the status, and what we encountered in practice is that very rarely we would get here with a status of can_be_merged, but the status changed in the moments between that successful response and the attempt to set the auto merge setting would result in an error (typically the 405 response documented here (https://docs.gitlab.com/ee/api/merge_requests.html#accept-mr).  I reproduced this scenario in one of the new tests, which fails without this block here.  

The mergeRequest is just returned here so we can proceed as normal having given up on the attempt to set the auto-merge on it.  Let me know if I can clarify it further!

Reply:
Just thinking about it a bit more, it might make it clearer to put this recoverWith directly on the client.put on L173 as that is the source of these potential error responses - happy to make that change.